acnportal/acnsim/simulator.py

```diff
@@ -184,7 +194,7 @@ def _store_actual_charging_rates(self):
         if self.iteration < self.charging_rates.shape[1]:
             self.charging_rates[:, self.iteration] = current_rates.T
         else:
-            self.charging_rates = _increase_width(self.charging_rates, self.event_queue.get_last_timestamp() + 1)
+            self.charging_rates = _increase_width(self.charging_rates, max(self.event_queue.get_last_timestamp() + 1, self._iteration + 1))
```

Comment:
Add comment for why self._iteration is necessary

Reply:
This actually wasn't necessary; but it was necessary to check that the event queue wasn't empty before doing this expansion. I've done this in the new version.